mesonbuild/interpreter/interpreter.py

```diff
@@ -1878,6 +1881,14 @@ def func_jar(self, node: mparser.BaseNode,
                  kwargs: kwtypes.Jar) -> build.Jar:
         return self.build_target(node, args, kwargs, build.Jar)
 
+    @permittedKwargs(build.known_shader_kwargs)
+    @typed_pos_args('shader', str, varargs=(str, mesonlib.File, build.CustomTarget, build.CustomTargetIndex, build.GeneratedList, build.ExtractedObjects, build.BuildTarget))
```

Comment:
Does the shader compiler support taking a C/C++ `.o` file extracted from another target?